### ISSUE TITLE
SyncReplicasOptimizer + MonitoredTrainingSession go through network many times with one session.run

### ISSUE THREAD

**Issue by muye5** [tensorflow/tensorflow#20834]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 2.7
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 7.0
- **GPU model and memory**: 16G
- **Exact command to reproduce**: None

### Describe the problem
training model by `tf.train.MonitoredTrainingSession` combined with `tf.train.SyncReplicasOptimizer`,
I find the model network will be run twice with only once `session.run` calling, and many NaN values will be raised. More, this will update the batch normalization parameters.
I do the initialization by `tf.train.SyncReplicasOptimizer.make_session_run_hook`. By the way, code will be right without `tf.train.SyncReplicasOptimizer`.

### Source code / logs
```
# 1. model define
input = tf.placeholder_with_default()
phase = tf.placeholder_with_default(True, shape=(), name='phase')
... network define ...
self.loss = xxx

# 2. optimizer define
self.optimizer = tf.train.AdamOptimizer(learning_rate=self.learning_rate)
self.optimizer = tf.train.SyncReplicasOptimizer(self.optimizer, replicas_to_aggregate=worker_num, total_num_replicas=worker_num）
update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
with tf.control_dependencies(update_ops):
     self.train_op = self.optimizer.minimize(self.loss, global_step=self.global_step)

# 3. hook define
sync_replicas_hook = self.optimizer.make_session_run_hook((flags.task_index == 0), num_tokens=0)

# 4. train
with tf.train.MonitoredTrainingSession(master=server.target, hooks=[sync_replicas_hook], is_chief=is_chief):
    outpus = sess.run([self.train_op, global_step], feed_dict={'phase:0':True, ...})
```

Problem A:
I add `tf.Print` line when define in network, I find this line executing twice with once sess.run calling.

Problem B: 
when `phase = True`, the BatchNorm Parameter update right, but when `phase = False`, `moving_mean` and `moving_variance` turn to NaN, so does the loss.

Everything is OK when training with async optimizer.

**Comment by muye5** [comment]
Another problem：
I am not sure whether someone has tried SyncReplicasOptimizer +  MonitoredTrainingSession + Placeholder of Model Input. After session created, during initializing variables in the hook function, it reports needing to feed value to the placeholder. I am confused what triggered off the network computing.

**Comment by bignamehyp** [comment]
This issue is likely due to custom network definition rather than a general one. Please provide details.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 31 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
